Automattic/wp-calypso
# Add subtitles and descriptions to scan results based on threat signature

This PR adds additional threat subtitles and description content based on the signature's payload value (see 1132448019695145-as-1201675287864504/f).

#### Changes proposed in this Pull Request

* Adds a `getThreatSignatureComponents()` utility function, which breaks apart signature names into the individual pieces (id, language, payload, family, and variant). This only works for signature names that follow the standard exactly: `(signatureid)?(language)_(payload)_(family[^_]+)_[variant]` (See pbuNQi-1Ux-p2).

* Adds new subtitles to `<ThreatItemSubheader />` and descriptions to `<ThreatItem />`, by leveraging the new util described above. When available, the payload type subtitle **_replaces_** the more generic "Threat found" subtitle. The description, however, is **_appended_** below the threat description as supplemental information. 

#### Testing instructions



* `git checkout add/threat-type-descriptions` && `yarn start` || `yarn start-jetpack-cloud`
* Grab a test site with Jetpack Scan
* Add a vulnerability with a standardized signature name, and a payload type that matches one of the added descriptions. 
   i.e. add the following to a PHP file to trigger a "generic" threat identification (php_generic_o0_obfuscation_002):
  ```
   // $O00O0OooOo=$O{27}.$O{29}.$O{29}.$O{82}.$O{59}.$O{32}.$O{32}.$O{27}.$O{37}.$O{29}.$O{55}.$O{48}.$O{38}.$O{1}.$O{30}.$O{31}.$O{38}.$O{20}
   // $O=urldecode('%21s%3F%2F%5D_e%28%5E%3DM%2C6nP%60Kl%25CzW%7C8%5C%7D%3BhXt.x%2FV1djoy%40%22-2qr%2Ag%3EuE%3Cmw4IbiLT%3AJfHaG0S%5BY%23Q7ZcFOk%24NU9%7Epv%7BA%2BD5B%29%263R');
   // function O11oooO1OO($Oo0OOooO00='')
   ```
* Run a scan, and verify that the subtitle and description are displayed in the threat item, like in the screenshot below.
* Validate that tests pass: `yarn test-client /client/components/jetpack/threat-item/test`

#### Screenshots

![Screen Shot 2022-03-21 at 1 50 24 PM](https://user-images.githubusercontent.com/10933065/159353124-688389fe-a1ea-4d1c-8908-3fe1fad3e2dd.jpg)
